### PR TITLE
Translated Component from Legacy APIs

### PR DESCRIPTION
**Pull request by zeuspsy0x1**
Added the translation of the Component md file, from the legacy APIs section. 

The route is src\content\reference\react\Component.md

1. I translated all the text information available that renders to the users and informs them about the Component section in the Legacy APIs. The titles are the same, but the explanations are now in spanish. 
2. Translated string messages in some parts of the code fragments, so the messages make more sense to spanish speakers. But I didn't change all the text words there because it's hard to doing it correctly without changing much of the original code fragments.

image

### CODE REVIEW

File: src/content/reference/react/Component.md [hunk 1]
```diff
@@ -26,33 +26,33 @@ class Greeting extends Component {
...
 
-[See more examples below.](#usage)
+[Mira más ejemplos.](#usage)
 
 ---
 
 ### `context` {/*context*/}
 
-The [context](/learn/passing-data-deeply-with-context) of a class component is available as `this.context`. It is only available if you specify *which* context you want to receive using [`static contextType`](#static-contexttype) (modern) or [`static contextTypes`](#static-contexttypes) (deprecated).
+El [contexto](/learn/passing-data-deeply-with-context) de un componente de clase está disponible como `this.context`. Solo está disponible si especificas *cual* contexto deseas recibir usando [static contextType](#static-contexttype) (moderno) or [`static contextTypes`](#static-contexttypes) (deprecado).
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:53]:
```suggestion
El [contexto](/learn/passing-data-deeply-with-context) de un componente de clase está disponible como `this.context`. Solo está disponible si especificas *cual* contexto deseas recibir usando [static contextType](#static-contexttype) (moderno) o [`static contextTypes`](#static-contexttypes) (deprecado).
```

File: src/content/reference/react/Component.md [hunk 2]
```diff
@@ -109,17 +109,17 @@ Reading `this.props` in class components is equivalent to [declaring props](/lea
 
 <Deprecated>
 
-This API will be removed in a future major version of React. [Use `createRef` instead.](/reference/react/createRef)
+Esta API será eliminada en una versión mayor futura de React. [Use createRef en su lugar.](/reference/react/createRef)
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:112]:
```suggestion
Esta API será eliminada en una versión mayor futura de React. [Usa createRef en su lugar.](/reference/react/createRef)
```

File: src/content/reference/react/Component.md [hunk 3]
```diff
@@ -148,17 +148,17 @@ class Counter extends Component {
...
+[Mira cómo migrar.](#migrating-a-component-with-state-from-a-class-to-a-function)
 
 </Note>
 
 ---
 
 ### `constructor(props)` {/*constructor*/}
 
-The [constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/constructor) runs before your class component *mounts* (gets added to the screen). Typically, a constructor is only used for two purposes in React. It lets you declare state and [bind](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) your class methods to the class instance:
+El [Constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/constructor) se ejecuta antes de que el componente de clase se *monte* (se agregue a la pantalla). Normalmente, en React se utiliza el constructor únicamente para dos propósitos. Te permite declarar el estado y [enlazar](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) los métodos de la clase con la instancia de la clase:
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:161]:
```suggestion
El [constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/constructor) se ejecuta antes de que el componente de clase se *monte* (se agregue a la pantalla). Normalmente, en React se utiliza el constructor únicamente para dos propósitos. Te permite declarar el estado y [enlazar](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) los métodos de la clase con la instancia de la clase:
```

File: src/content/reference/react/Component.md [hunk 4]
```diff
@@ -184,73 +184,73 @@ class Counter extends Component {
...
 
-A constructor should not contain any side effects or subscriptions.
+Un constructor no debería tener efectos secundarios ni suscripciones.
 
-#### Parameters {/*constructor-parameters*/}
+#### Parámetros {/*constructor-parameters*/}
 
-* `props`: The component's initial props.
+* `props`: Las props iniciales del componente.
+#### Retorna {/*constructor-returns*/}
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:192]:
```suggestion

#### Retorna {/*constructor-returns*/}
```

File: src/content/reference/react/Component.md [hunk 5]
```diff
@@ -184,73 +184,73 @@ class Counter extends Component {
...
 
-#### Returns {/*constructor-returns*/}
+El `constructor` no debe retornar nada.
 
-`constructor` should not return anything.
+#### Precauciones {/*constructor-caveats*/}
 
-#### Caveats {/*constructor-caveats*/}
+* No ejecute ningún efecto secundario o suscripciones en el constructor. En su lugar, utilice [`componentDidMount`](#componentdidmount) para eso.
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:198]:
```suggestion
* No ejecutes ningún efecto secundario o suscripciones en el constructor. En su lugar, utiliza [`componentDidMount`](#componentdidmount) para eso.
```

File: src/content/reference/react/Component.md [hunk 6]
```diff
@@ -184,73 +184,73 @@ class Counter extends Component {
...
 
-* Do not run any side effects or subscriptions in the constructor. Instead, use [`componentDidMount`](#componentdidmount) for that.
+* Dentro de un constructor, debes llamar a `super(props)` antes que cualquier otra declaración. Si no lo haces, `this.props` será `undefined` mientras se ejecuta el constructor, lo que puede ser confuso y causar errores.
 
-* Inside a constructor, you need to call `super(props)` before any other statement. If you don't do that, `this.props` will be `undefined` while the constructor runs, which can be confusing and cause bugs.
+* El constructor es el único lugar donde puedes asignar [`this.state`](#state) directamente. En todos los demás métodos, debes utilizar [`this.setState ()`](#setstate) en su lugar. No llames a `setState` en el constructor.
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:202]:
```suggestion
* El constructor es el único lugar donde puedes asignar [`this.state`](#state) directamente. En todos los demás métodos, debes utilizar [`this.setState()`](#setstate) en su lugar. No llames a `setState` en el constructor.
```

File: src/content/reference/react/Component.md [hunk 7]
```diff
@@ -340,45 +340,44 @@ class ChatRoom extends Component {
...
 
 <Note>
 
-For many use cases, defining `componentDidMount`, `componentDidUpdate`, and `componentWillUnmount` together in class components is equivalent to calling [`useEffect`](/reference/react/useEffect) in function components. In the rare cases where it's important for the code to run before browser paint, [`useLayoutEffect`](/reference/react/useLayoutEffect) is a closer match.
-
-[See how to migrate.](#migrating-a-component-with-lifecycle-methods-from-a-class-to-a-function)
+Para muchos casos de uso, definir juntos `componentDidMount`, `componentDidUpdate` y `componentWillUnmount` en componentes de clase es equivalente a llamar a [`useEffect`](/reference/react/useEffect) en componentes de función. En los casos raros donde es importante que el código se ejecute antes de renderizar en el navegador, [`useLayoutEffect`](/reference/react/useLayoutEffect) es una opción más cercana.
 
+[Consulta cómo migrar](#migrating-a-component-with-lifecycle-methods-from-a-class-to-a-function).
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:370]:
```suggestion
[Mira cómo migrar](#migrating-a-component-with-lifecycle-methods-from-a-class-to-a-function).
```

Para mantener consistencia con el resto de los ejemplos

File: src/content/reference/react/Component.md [hunk 8]
```diff
@@ -340,45 +340,44 @@ class ChatRoom extends Component {
...
 </Note>
 ---
 
 ### `componentWillMount()` {/*componentwillmount*/}
 
 <Deprecated>
 
-This API has been renamed from `componentWillMount` to [`UNSAFE_componentWillMount`.](#unsafe_componentwillmount) The old name has been deprecated. In a future major version of React, only the new name will work.
+Este API ha sido renombrado de `componentWillMount` a [`UNSAFE_componentWillMount`.](#unsafe_componentwillmount) El nombre antiguo ha sido deprecado. En una versión importante futura de React, solo funcionará el nuevo nombre.
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:378]:
```suggestion
Esta API ha sido renombrado de `componentWillMount` a [`UNSAFE_componentWillMount`.](#unsafe_componentwillmount) El nombre antiguo ha sido deprecado. En una versión importante futura de React, solo funcionará el nuevo nombre.
```

File: src/content/reference/react/Component.md [hunk 9]
```diff
@@ -400,19 +400,19 @@ Run the [`rename-unsafe-lifecycles` codemod](https://github.com/reactjs/react-co
 
 <Deprecated>
 
-This API has been renamed from `componentWillUpdate` to [`UNSAFE_componentWillUpdate`.](#unsafe_componentwillupdate) The old name has been deprecated. In a future major version of React, only the new name will work.
+Esta API ha sido renombrada de `componentWillUpdate` a [`UNSAFE_componentWillUpdate`.](#unsafe_componentwillupdate) El nombre antiguo ha sido desaprobado. En una versión mayor futura de React, solo funcionará el nuevo nombre.
 
-Run the [`rename-unsafe-lifecycles` codemod](https://github.com/reactjs/react-codemod#rename-unsafe-lifecycles) to automatically update your components.
+Ejecuta [`rename-unsafe-lifecycles` codemod](https://github.com/reactjs/react-codemod#rename-unsafe-lifecycles) para actualizar automáticamente sus componentes.
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:405]:
```suggestion
Ejecuta [`rename-unsafe-lifecycles` codemod](https://github.com/reactjs/react-codemod#rename-unsafe-lifecycles) para actualizar automáticamente tus componentes.
```

File: src/content/reference/react/Component.md [hunk 10]
```diff
@@ -734,147 +729,145 @@ class Rectangle extends Component {
...
+Devuelve `false` para indicar a React que se puede omitir la re-renderización.
 
-#### Caveats {/*shouldcomponentupdate-caveats*/}
+#### Precauciones {/*shouldcomponentupdate-caveats*/}
 
-- This method *only* exists as a performance optimization. If your component breaks without it, fix that first. 
+- Este método solo existe como una optimización de rendimiento. Si su componente falla sin él, primero solucione eso.
 
-- Consider using [`PureComponent`](/reference/react/PureComponent) instead of writing `shouldComponentUpdate` by hand. `PureComponent` shallowly compares props and state, and reduces the chance that you'll skip a necessary update.
+- Considere usar [`PureComponent`](/reference/react/PureComponent) en lugar de escribir `shouldComponentUpdate` manualmente. `PureComponent` compara superficialmente las props y el estado, y reduce la posibilidad de que omita una actualización necesaria.
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:750]:
```suggestion
- Considera usar [`PureComponent`](/reference/react/PureComponent) en lugar de escribir `shouldComponentUpdate` manualmente. `PureComponent` compara superficialmente las props y el estado, y reduce la posibilidad de que omita una actualización necesaria.
```

File: src/content/reference/react/Component.md [hunk 11]
```diff
@@ -1064,46 +1032,47 @@ class Form extends Component {
...
 ---
 
-## Usage {/*usage*/}
+## Uso {/*usage*/}
 
-### Defining a class component {/*defining-a-class-component*/}
+### Definiendo un componente de clase {/*defining-a-class-component*/}
 
-To define a React component as a class, extend the built-in `Component` class and define a [`render` method:](#render)
+Para definir un componente React como una clase, extiende la clase integrada `Component` y define un método [`render:`](#render).
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:1075]:
```suggestion
Para definir un componente React como una clase, extiende la clase integrada `Component` y define un [`método render:`](#render).
```

File: src/content/reference/react/Component.md [hunk 12]
```diff
@@ -1335,21 +1304,21 @@ button { margin-left: 10px; }
...
 
-We recommend defining components as functions instead of classes. [See how to migrate.](#migrating-a-component-with-lifecycle-methods-from-a-class-to-a-function)
+Recomendamos definir los componentes como funciones en lugar de clases. [Mira cómo migrar](#migrating-a-component-with-lifecycle-methods-from-a-class-to-a-function).
 
 </Pitfall>
 
 ---
 
-### Catching rendering errors with an error boundary {/*catching-rendering-errors-with-an-error-boundary*/}
+### Capturando errores de renderizado con un límite de errores {/*catching-rendering-errors-with-an-error-boundary*/}
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:1317]:
```suggestion
### Capturando errores de renderizado con un error boundary {/*catching-rendering-errors-with-an-error-boundary*/}
```

File: src/content/reference/react/Component.md [hunk 13]
```diff
@@ -1383,33 +1352,33 @@ class ErrorBoundary extends React.Component {
...
 
-You don't need to wrap every component into a separate error boundary. When you think about the [granularity of error boundaries,](https://aweary.dev/fault-tolerance-react/) consider where it makes sense to display an error message. For example, in a messaging app, it makes sense to place an error boundary around the list of conversations. It also makes sense to place one around every individual message. However, it wouldn't make sense to place a boundary around every avatar.
+No es necesario envolver cada componente en un error boundary por separado. Al considerar la [granularidad de los límites de errores,](https://aweary.dev/fault-tolerance-react/) considere dónde tiene sentido mostrar un mensaje de error. Por ejemplo, en una aplicación de mensajería, tiene sentido colocar un error boundary alrededor de la lista de conversaciones. También tiene sentido colocar uno alrededor de cada mensaje individual. Sin embargo, no tendría sentido colocar un límite alrededor de cada avatar.
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:1365]:
```suggestion
No es necesario envolver cada componente en un error boundary por separado. Al considerar la [granularidad de los error boundaries,](https://aweary.dev/fault-tolerance-react/) considere dónde tiene sentido mostrar un mensaje de error. Por ejemplo, en una aplicación de mensajería, tiene sentido colocar un error boundary alrededor de la lista de conversaciones. También tiene sentido colocar uno alrededor de cada mensaje individual. Sin embargo, no tendría sentido colocar un límite alrededor de cada avatar.
```

File: src/content/reference/react/Component.md [hunk 14]
```diff
@@ -673,41 +668,41 @@ You can also pass a function to `setState`. It lets you update state based on th
...
+- `setState` no actualiza `this.state` inmediatamente. Esto puede ser un problema potencial si se lee `this.state` justo después de llamar a `setState`. En su lugar, utiliza [`componentDidUpdate`](#componentdidupdate) o el argumento de `callback` de setState, ya que ambos están garantizados para activarse después de que se haya aplicado la actualización. Si necesitas establecer el estado basado en el estado anterior, puedes pasar una función a `nextState` como se describe anteriormente.
 
 <Note>
 
-Calling `setState` in class components is similar to calling a [`set` function](/reference/react/useState#setstate) in function components.
+Llamar a `setState` en componentes de clase es similar a llamar a una función `set` en componentes funcionales.
 
-[See how to migrate.](#migrating-a-component-with-state-from-a-class-to-a-function)
+[Ver cómo migrar.](#migrating-a-component-with-state-from-a-class-to-a-function)
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:695]:
```suggestion
[Mira cómo migrar.](#migrating-a-component-with-state-from-a-class-to-a-function)
```

File: src/content/reference/react/Component.md [hunk 15]
```diff
@@ -926,19 +919,19 @@ class Button extends Component {
 
 <Note>
 
-Reading `this.context` in class components is equivalent to [`useContext`](/reference/react/useContext) in function components.
+Leer `this.context` en componentes de clase es equivalente a [`useContext`](/reference/react/useContext) en componentes de función.
 
-[See how to migrate.](#migrating-a-component-with-context-from-a-class-to-a-function)
+[Ver cómo migrar.](#migrating-a-component-with-context-from-a-class-to-a-function)
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:924]:
```suggestion
[Mira cómo migrar.](#migrating-a-component-with-context-from-a-class-to-a-function)
```

File: src/content/reference/react/Component.md [hunk 16]
```diff
@@ -137,9 +137,9 @@ class Counter extends Component {
     return (
       <>
         <button onClick={this.handleAgeChange}>
-        Increment age
+        Incrementar edad
         </button>
-        <p>You are {this.state.age}.</p>
+        <p>Tienes {this.state.age}.</p>
```
Comment (mateoguzmana) [inline src/content/reference/react/Component.md:142]:
```suggestion
        <p>Tienes {this.state.age} años.</p>
```

**Comment by zeuspsy0x1** [comment]
> 

Gracias Mateo, se me habían pasado muchos errores. Ahí ya hice commit de tus sugerencias.

**Review by mateoguzmana — APPROVED** [review]

**Pull request merged**